### PR TITLE
udf: rework IFile and IFileDirectory implementation

### PR DESCRIPTION
**Pull request by lrusak**
This change allows using libcdio's libudf

we already build libudf as part of the depends so let's use it.

Again, this code isn't really used for much unless reading a raw iso that contains a udf filesystem. For dvd's libdvd will open it not this code. So unless you open an iso that contains a udf filesystem and contains files not in a dvd file structure you won't be using this code.

I tested this using udf iso's containing mp3 files that was made with `mkisofs`

This gets rid of windows api's as part of a larger cleanup.

### CODE REVIEW

**Comment by lrusak** [comment]
not really sure about the build error 🤕 

```
14:24:54 /home/jenkins/android-dev//android-toolchain-arm-21-r18/bin/../lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: error: ../tools/depends/xbmc-depends/arm-linux-androideabi-21-debug/lib/libcdio.a(logging.o): multiple definition of 'cdio_warn'
14:24:54 /home/jenkins/android-dev//android-toolchain-arm-21-r18/bin/../lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: build/filesystem/filesystem.a(iso9660.cpp.o): previous definition here
```

```
13:52:14 duplicate symbol _cdio_warn in:
13:52:14     build/filesystem/filesystem.a(iso9660.cpp.o)
13:52:14     ../tools/depends/xbmc-depends/iphoneos11.0_arm64-target-debug/lib/libcdio.a(logging.o)
```

File: cmake/modules/FindUdf.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# and the following imported targets::
+#
+#   UDF::UDF - The udf library
+
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_UDF libudf>=0.94 QUIET)
+endif()
+
+find_path(UDF_INCLUDE_DIR NAMES cdio/udf.h
+                          PATHS ${PC_UDF_libudf_INCLUDEDIR})
```
Comment (Rechi) [inline cmake/modules/FindUdf.cmake]: PC_UDF_INCLUDEDIR

File: cmake/modules/FindUdf.cmake [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+find_library(UDF_LIBRARY NAMES udf libudf
+                         PATHS ${PC_UDF_libudf_LIBDIR})
```
Comment (Rechi) [inline cmake/modules/FindUdf.cmake]: PC_UDF_LIBDIR

File: cmake/modules/FindUdf.cmake [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+set(UDF_VERSION ${PC_UDF_libudf_VERSION})
```
Comment (Rechi) [inline cmake/modules/FindUdf.cmake]: PC_UDF_VERSION

File: cmake/modules/FindUdf.cmake [hunk 4]
```diff
@@ -0,0 +1,45 @@
+#.rst:
+# FindCdio
```
Comment (Rechi) [inline cmake/modules/FindUdf.cmake]: Udf

File: xbmc/filesystem/UDFDirectory.cpp [hunk 5]
```diff
@@ -8,55 +8,67 @@
...
-
-CUDFDirectory::CUDFDirectory(void) = default;
+#include <cdio/udf.h>
 
-CUDFDirectory::~CUDFDirectory(void) = default;
+using namespace XFILE;
 
-bool CUDFDirectory::GetDirectory(const CURL& url,
-                                 CFileItemList &items)
+bool CUDFDirectory::GetDirectory(const CURL& url, CFileItemList &items)
```
Comment (Rechi) [inline xbmc/filesystem/UDFDirectory.cpp]: left align `&`

**Review by Rechi — COMMENTED** [review]
new required build dependency is missing at least in docs/README.Linux.md, docs/README.Ubuntu.md and .travis.yml

**Pull request merged**